### PR TITLE
79 misc 14 bug fixing for database and gamecontroller

### PR DESCRIPTION
**Pull request by SMateen21**
Updated gamecontroller to fix a bug with attribute name (selectedOwner -> currentPlayer). Found CSVDatabase did not update wins and losses if a player was in the database, fixed that bug by using a CSV editing library called openCSV (added dependency to gradle build). Emptied players.csv as to have a clean slate for main.

### CODE REVIEW

File: src/main/java/controllers/GameCreationController.java [hunk 1]
```diff
@@ -34,11 +34,21 @@ public boolean createGameResponse(ArrayList<PlayerCreationInformation> playersIn
...
-        if(requestInfo.keySet().size() != playersInfo.size()) {
+        if(requestInfo.keySet().size() != playersInfo.size() | !(validOptions(requestInfo))) {
             return false;
         }
         GameCreationRequestModel request = new GameCreationRequestModel(requestInfo);
         GameCreationResponseModel response = inputBoundary.createGameResponse(request);
         return response.getGameCreated();
     }
+
+    private boolean validOptions(HashMap<String, Boolean> info) {
```
Comment (meiiph) [inline src/main/java/controllers/GameCreationController.java:45]: Javadocs please :)

File: src/main/java/database/CSVDatabase.java [hunk 2]
```diff
@@ -45,18 +50,39 @@ public boolean savePlayer(HumanPlayer player) throws IOException {
...
         String[] data = {player.getName(), wins, losses};
         String line = String.join(",", data);
         bw.write(line);
-        bw.newLine();
+        bw.write("\n");
         bw.close();
         return true;
     }
 
+    private void updatePlayerStats(PlayerInformation playerInfo) {
```
Comment (meiiph) [inline src/main/java/database/CSVDatabase.java:66]: Javadocs please :)

**Review by meiiph — APPROVED** [review]
Looks good, just add javadocs to the two places requested, and it'll be ready for approval!

**Review by lilyxmeng — APPROVED** [review]

File: build.gradle [hunk 3]
```diff
@@ -15,6 +15,7 @@ repositories {
 
 dependencies {
     implementation 'junit:junit:4.13.1'
+    implementation 'com.opencsv:opencsv:5.8'
```
Comment (nyan-ko) [inline build.gradle:18]: have you verified this works on a fresh install?

**Review by nyan-ko — APPROVED** [review]
approving to test opencsv install

**Review by meiiph — DISMISSED** [review]
My comments have been addressed.👍

**Review by nyan-ko — DISMISSED** [review]

**Review by lilyxmeng — APPROVED** [review]

**Pull request merged**